core/modules/replica/WorkerSqlRequest.cc

```diff
@@ -0,0 +1,217 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/WorkerSqlRequest.h"
+
+// System headers
+#include <stdexcept>
+
+// Qserv headers
+#include "replica/Configuration.h"
+#include "replica/DatabaseMySQL.h"
+#include "replica/Performance.h"
+#include "util/BlockPost.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+using namespace std;
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.WorkerSqlRequest");
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+WorkerSqlRequest::Ptr WorkerSqlRequest::create(ServiceProvider::Ptr const& serviceProvider,
+                                               string const& worker,
+                                               string const& id,
+                                               int priority,
+                                               std::string const& query,
+                                               std::string const& user,
+                                               std::string const& password,
+                                               size_t maxRows) {
+    return WorkerSqlRequest::Ptr(
+        new WorkerSqlRequest(serviceProvider,
+                             worker,
+                             id,
+                             priority,
+                             query,
+                             user,
+                             password,
+                             maxRows));
+}
+
+
+WorkerSqlRequest::WorkerSqlRequest(ServiceProvider::Ptr const& serviceProvider,
+                                   string const& worker,
+                                   string const& id,
+                                   int priority,
+                                   std::string const& query,
+                                   std::string const& user,
+                                   std::string const& password,
+                                   size_t maxRows)
+    :   WorkerRequest(serviceProvider,
+                      worker,
+                      "SQL",
+                      id,
+                      priority),
+        _query(query),
+        _user(user),
+        _password(password),
+        _maxRows(maxRows) {
+}
+
+
+void WorkerSqlRequest::setInfo(ProtocolResponseSql& response) const {
+
+    LOGS(_log, LOG_LVL_DEBUG, context(__func__));
+
+    util::Lock lock(_mtx, context(__func__));
+
+    // Update the performance of the target request before returning it
+    response.set_allocated_target_performance(performance().info());
+
+    // Carry over the result of the query only after the request
+    // has finished (or failed).
+    switch (status()) {
+        case STATUS_SUCCEEDED:
+        case STATUS_FAILED:
```

Comment:
DO you need `fall-through` comment here to suppress warnings?

Reply:
Ot if the `case` is empty. The complier complains for non-trivial scenarios only, such as:
```
switch(val) {
    case 123:
        a = 123;
    case 345:
        break;